torch/fx/experimental/proxy_tensor.py

```diff
@@ -70,7 +72,64 @@ def decompose(decomposition_table):
 proxy_slot = object()
 no_default = object()
 
+
 py_sym_types = (SymInt, SymFloat, SymBool)
+
+
+class SuppressGuardsAndCheckIfAnyAdded:
```

Comment:
(1) This should be in shape_env 
(2) Where are you suppressing? by trimming in exit? That isn't sound or right to do
(3) This entirely ignores how produce_guards works
(4) There's already a _suppress_guards_tls

Reply:
> That isn't sound or right to do
Can you explain why ? I am admittedly ignorant

> (3) This entirely ignores how produce_guards works

How so ? 

> `_suppress_guards_tls` 

This doesn tdo the same thing.


I can switch to just checking if the expressions are equal, i just would like to know why this doesnt work.